main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1343,7 +1343,12 @@ object Typer {
           (constrs, tpe, pur, eff) <- visitExp(exp)
         } yield (constrs, tpe, Type.Pure, Type.Empty)
 
-      case KindedAst.Expression.Upcast(exp, tvar, loc) =>
+      case KindedAst.Expression.Upcast(exp, tvar, _) =>
+        for {
+          (constrs, _, pur, eff) <- visitExp(exp)
```

Comment:
Can we add a comment for each of these to point out the reason we are ignoring the type of `exp`